commit 989e25ecc4dc75c842f13b2b6457df9a3533ac72
Author: heitbaum <rudi@heitbaum.com>
Date:   Sun Feb 6 07:56:38 2022 +0000

    meson: enable override of arch-independent pkg-config
    
    Add meson_option pkgconfigdatadir to set the directory for
    arch-independent pkg-config files.
    
    Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>

diff --git a/meson.build b/meson.build
index 5c15cc9..bb9c7c1 100644
--- a/meson.build
+++ b/meson.build
@@ -57,12 +57,14 @@ pc_data = configuration_data()
 pc_data.set('prefix', get_option('prefix'))
 # meson does not allow installing the includedir outside of the prefix
 pc_data.set('includedir', '${prefix}/' + get_option('includedir'))
+# Dirs of external packages
+pkgconfigdatadir = get_option('pkgconfigdatadir') != '' ? get_option('pkgconfigdatadir') : get_option('datadir') / 'pkgconfig'
 
 foreach pc : pcs
     configure_file(
         input : pc + '.pc.in',
         output : pc + '.pc',
-        install_dir : get_option('datadir') + '/pkgconfig',
+        install_dir : pkgconfigdatadir,
         configuration : pc_data,
     )
 endforeach
@@ -100,7 +102,7 @@ if get_option('legacy') == true
         configure_file(
             input : pc + '.pc.in',
             output : pc + '.pc',
-            install_dir : get_option('datadir') + '/pkgconfig',
+            install_dir : pkgconfigdatadir,
             configuration : pc_data,
         )
     endforeach
diff --git a/meson_options.txt b/meson_options.txt
index 757cd72..4c80f32 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1 +1,3 @@
 option('legacy', type: 'boolean', value: false)
+option('pkgconfigdatadir', type : 'string', value : '',
+       description : 'directory for arch-independent pkg-config files')
